sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -348,34 +344,24 @@ case class AdaptiveSparkPlanExec(
     withFinalPlanUpdate(_.execute())
   }
 
-  /**
-   * Determine if the final query stage supports columnar execution. Calling this method
-   * will trigger query execution of child query stages if they have not already executed.
-   *
-   * If this method returns true then it is safe to call doExecuteColumnar to execute the
-   * final stage.
-   */
-  def finalPlanSupportsColumnar(): Boolean = {
-    getFinalPhysicalPlan().supportsColumnar
-  }
-
   override def doExecuteColumnar(): RDD[ColumnarBatch] = {
     withFinalPlanUpdate(_.executeColumnar())
   }
 
+  override def doExecuteBroadcast[T](): broadcast.Broadcast[T] = {
```

Comment:
can we move this method back?

Reply:
it's not just moving around, it follows the other methods to also call `withFinalPlanUpdate`.